recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,197 @@
+    #             return '.x86_64'
+    #     return ""
+    #
+    # def build_requirements(self):
+    #     pack_names = None
+    #     if not self.options.minimal and tools.os_info.is_linux:
+    #         if tools.os_info.with_apt:
+    #             pack_names = [
+    #                 "freeglut3-dev",
+    #                 "mesa-common-dev",
+    #                 "mesa-utils-extra",
+    #                 "libgl1-mesa-dev",
+    #                 "libglapi-mesa",
+    #                 "libsm-dev",
+    #                 "libx11-dev",
+    #                 "libxext-dev",
+    #                 "libxt-dev",
+    #                 "libglu1-mesa-dev"]
+    #
+    #     if pack_names:
+    #         installer = tools.SystemPackageTool()
+    #         for item in pack_names:
+    #             installer.install(item + self._system_package_architecture())
+    
+    def requirements(self):
+        # 1) Is it correct that "opengl/system" is required for every OS?
+        # 2) Is it correct that "opengl/system" is even when compiled without Qt?
+        self.requires("opengl/system")
+        if self.settings.os == "Linux":
+            if self.options.x11:
+                # Do we need "xorg/system"?
+                self.requires("xorg/system")
+        if self.options.with_qt:
+            # FIXME: Missing qt recipe. Qt recipe PR: https://github.com/conan-io/conan-center-index/pull/1759
+            # When qt available, "self.requires("qt/5.15.1")" should replace below line.
+            raise ConanInvalidConfiguration("VTK option 'with_qt' requires 'qt:shared=True'")
+                
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = "OFF"
+        self._cmake.definitions["BUILD_EXAMPLES"] = "OFF"
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+
+        if self.options.minimal:
+            self._cmake.definitions["VTK_Group_StandAlone"] = "OFF"
+            self._cmake.definitions["VTK_Group_Rendering"] = "OFF"
+        if self.options.ioxml:
+            self._cmake.definitions["Module_vtkIOXML"] = "ON"
+        if self.options.ioexport:
+            self._cmake.definitions["Module_vtkIOExport"] = "ON"
+        if self.options.qt:
+            self._cmake.definitions["VTK_Group_Qt"] = "ON"
+            self._cmake.definitions["VTK_MODULE_ENABLE_VTK_GUISupportQt"] = "YES"
+            self._cmake.definitions["VTK_MODULE_ENABLE_VTK_GUISupportQtOpenGL"] = "YES"
+            self._cmake.definitions["VTK_MODULE_ENABLE_VTK_RenderingQt"] = "YES"
+        if self.options.mpi:
+            self._cmake.definitions["VTK_Group_MPI"] = "ON"
+            self._cmake.definitions["Module_vtkIOParallelXML"] = "ON"
+        if self.options.mpi_minimal:
+            self._cmake.definitions["Module_vtkIOParallelXML"] = "ON"
+            self._cmake.definitions["Module_vtkParallelMPI"] = "ON"
```

Comment:
Check CMake's [`if` documtation](https://cmake.org/cmake/help/latest/command/if.html).
`ON`, `YES`, `TRUE` are interpreted as true-ish (and are case insensitive),
`OFF`, `NO`, `FALSE` are interpreted as false-ish.

Because these are used in a method named `_configure_cmake`, you can assume that these used as arguments to cmake.

I made the following suggestion to always set the definitions.
Doing it this way, makes sure that the option is set to a particular value (instead of auto-detection).

```suggestion
        on_off = lambda v: "ON" if v else "OFF"
        yes_no = lambda v: "YES" if v else "NO"
        self._cmake.definitions["VTK_Group_StandAlone"] = on_off(not self.options.minimal)
        self._cmake.definitions["VTK_Group_Rendering"] = on_off(not self.options.minimal)
        self._cmake.definitions["Module_vtkIOXML"] = on_off(self.options.ioxml)
        self._cmake.definitions["Module_vtkIOExport"] = on_off(self.options.ioexport)
        self._cmake.definitions["VTK_Group_Qt"] = on_off(self.options.qt)
        self._cmake.definitions["VTK_MODULE_ENABLE_VTK_GUISupportQt"] = yes_no(self.options.qt)
        self._cmake.definitions["VTK_MODULE_ENABLE_VTK_GUISupportQtOpenGL"] = yes_no(self.options.qt)
        self._cmake.definitions["VTK_MODULE_ENABLE_VTK_RenderingQt"] = yes_no(self.options.qt)
        self._cmake.definitions["VTK_Group_MPI"] = on_off(self.options.mpi)
        self._cmake.definitions["Module_vtkIOParallelXML"] = on_off(self.options.mpi)
        self._cmake.definitions["Module_vtkIOParallelXML"] = on_off(self.options.mpi_minimal)
        self._cmake.definitions["Module_vtkParallelMPI"] = on_off(self.options.mpi_minimal)
```

Reply:
Ah... Right, right, right!!
Done